src/core/server/elasticsearch/version_check/ensure_es_version.test.ts

```diff
@@ -148,6 +171,100 @@ describe('pollEsNodesVersion', () => {
       });
   });
 
+  it('returns the error from a failed nodes.info call when a poll request throws', (done) => {
+    expect.assertions(2);
+    const expectedCompatibilityResults = [false];
+    const expectedMessageResults = [
+      'Unable to retrieve version information from Elasticsearch nodes. mock request error',
+    ];
+    jest.clearAllMocks();
+
+    nodeInfosErrorOnce('mock request error');
+
+    pollEsNodesVersion({
+      internalClient,
+      esVersionCheckInterval: 1,
+      ignoreVersionMismatch: false,
+      kibanaVersion: KIBANA_VERSION,
+      log: mockLogger,
+    })
+      .pipe(take(1))
+      .subscribe({
+        next: (result) => {
+          expect(result.isCompatible).toBe(expectedCompatibilityResults.shift());
+          expect(result.message).toBe(expectedMessageResults.shift());
+        },
+        complete: done,
+        error: done,
+      });
+  });
+
+  it('only emits if the error from a failed nodes.info call changed from the previous poll', (done) => {
+    expect.assertions(4);
+    const expectedCompatibilityResults = [false, false];
+    const expectedMessageResults = [
+      'Unable to retrieve version information from Elasticsearch nodes. mock request error',
+      'Unable to retrieve version information from Elasticsearch nodes. mock request error 2',
+    ];
+    jest.clearAllMocks();
+
+    nodeInfosErrorOnce('mock request error'); // emit
+    nodeInfosErrorOnce('mock request error'); // ignore, same error message
+    nodeInfosErrorOnce('mock request error 2'); // emit
+
+    pollEsNodesVersion({
+      internalClient,
+      esVersionCheckInterval: 1,
+      ignoreVersionMismatch: false,
+      kibanaVersion: KIBANA_VERSION,
+      log: mockLogger,
+    })
+      .pipe(take(2))
+      .subscribe({
+        next: (result) => {
+          expect(result.message).toBe(expectedMessageResults.shift());
+          expect(result.isCompatible).toBe(expectedCompatibilityResults.shift());
+        },
+        complete: done,
+        error: done,
+      });
+  });
+
+  it('returns isCompatible=false and keeps polling when a poll request throws, only responding again if the error message has changed', (done) => {
+    expect.assertions(8);
+    const expectedCompatibilityResults = [false, false, true, false];
+    const expectedMessageResults = [
+      'This version of Kibana (v5.1.0) is incompatible with the following Elasticsearch nodes in your cluster: v5.0.0 @ http_address (ip)',
+      'Unable to retrieve version information from Elasticsearch nodes. mock request error',
+      "You're running Kibana 5.1.0 with some different versions of Elasticsearch. Update Kibana or Elasticsearch to the same version to prevent compatibility issues: v5.2.0 @ http_address (ip), v5.1.1-Beta1 @ http_address (ip)",
+      'Unable to retrieve version information from Elasticsearch nodes. mock request error 2',
```

Comment:
NIT: can we make this error the same as in line 239 to prove that the same error also emits after having switched earlier to a success reponse?